### DISCUSSION TITLE
Plans for Journey's End?

### DISCUSSION THREAD

**Discussion by AmphyEX** [Pryaxis/TShock#1704] [General]
I am unfamiliar with tShock and how terraria mods/plugins work, but are there plans to update tShock when the new update comes out? Do you think the update will be easy or difficult to do? I am hoping to host a Terraria server when the new update comes out, but I don't know if I want to do it before tShock updates.

**Comment by QuiCM** [comment]
We plan to update once 1.4 launches. At this point we can't provide a timeline for when the update will be available

**Reply by kubedzero** [reply]
Any guidance on if players running 1.4 or worlds generated with 1.4 will be connectable/loadable to the current version of TShock (4.3.26 running Terraria server 1.3.5.3)?

**Reply by hakusaro** [reply]
@kubedzero TShock loads Terraria Server. If the native server will allow you to run a prior world, TShock will.

_We don't know if backwards compatibility will still be a thing, but it probably will be._

**Reply by bartico6** [reply]
```Any guidance on if players running 1.4 or worlds generated with 1.4 will be connectable/loadable to the current version of TShock (4.3.26 running Terraria server 1.3.5.3)?```
Worlds generated using 1.4 will probably not load into 1.3.5.3 without causing a heap of exceptions. I'm also certain 1.4 players won't be able to join 1.3.5.3 servers either.

**Reply by hakusaro** [reply]
Ah, I totally misread the question. Yes, you won't be able to go backwards.

**Comment by moisterrific** [comment]
If you plan on hosting a server open to the public then I highly suggest waiting for it to be updated, because TShock comes with a bunch of anti-cheat, anti-grief, and plugin support that are vital for a server for public use. Otherwise if you just plan on playing with trusted friends you’ll be fine with the vanilla Terraria server that Relogic releases alongside 1.4. 

Whether plugins are updated for the new TShock is completely up to the discretion of the plugin developers.
